third_party/asr/asr582x/BUILD.gn

```diff
@@ -0,0 +1,382 @@
+        "lwip/mbedtls-mbedtls-2.18/library/md.c",
+        "lwip/mbedtls-mbedtls-2.18/library/md2.c",
+        "lwip/mbedtls-mbedtls-2.18/library/md4.c",
+        "lwip/mbedtls-mbedtls-2.18/library/md5.c",
+        "lwip/mbedtls-mbedtls-2.18/library/md_wrap.c",
+        "lwip/mbedtls-mbedtls-2.18/library/memory_buffer_alloc.c",
+        "lwip/mbedtls-mbedtls-2.18/library/nist_kw.c",
+        "lwip/mbedtls-mbedtls-2.18/library/oid.c",
+        "lwip/mbedtls-mbedtls-2.18/library/padlock.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pem.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pk.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pk_wrap.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pkcs11.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pkcs12.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pkcs5.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pkparse.c",
+        "lwip/mbedtls-mbedtls-2.18/library/pkwrite.c",
+        "lwip/mbedtls-mbedtls-2.18/library/platform.c",
+        "lwip/mbedtls-mbedtls-2.18/library/platform_util.c",
+        "lwip/mbedtls-mbedtls-2.18/library/poly1305.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ripemd160.c",
+        "lwip/mbedtls-mbedtls-2.18/library/rsa.c",
+        "lwip/mbedtls-mbedtls-2.18/library/rsa_internal.c",
+        "lwip/mbedtls-mbedtls-2.18/library/sha1.c",
+        "lwip/mbedtls-mbedtls-2.18/library/sha256.c",
+        "lwip/mbedtls-mbedtls-2.18/library/sha512.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_cache.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_ciphersuites.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_cli.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_cookie.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_srv.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_ticket.c",
+        "lwip/mbedtls-mbedtls-2.18/library/ssl_tls.c",
+        "lwip/mbedtls-mbedtls-2.18/library/threading.c",
+        "lwip/mbedtls-mbedtls-2.18/library/timing.c",
+        "lwip/mbedtls-mbedtls-2.18/library/version.c",
+        "lwip/mbedtls-mbedtls-2.18/library/version_features.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509_create.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509_crl.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509_crt.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509_csr.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509write_crt.c",
+        "lwip/mbedtls-mbedtls-2.18/library/x509write_csr.c",
+        "lwip/mbedtls-mbedtls-2.18/library/xtea.c",
+        "lwip/mbedtls-mbedtls-2.18/asr_mbedtls/alt/entropy_hardware_alt.c",
+        "lwip/mbedtls-mbedtls-2.18/asr_mbedtls/alt/aes_alt.c",
+        "lwip/mbedtls-mbedtls-2.18/asr_mbedtls/alt/sha1_alt.c",
+        "lwip/mbedtls-mbedtls-2.18/asr_mbedtls/alt/sha256_alt.c",
+  ]
+
+    foreach(src, _sources) {
+      sources += [ rebase_path(src,
+                              "${chip_root}/third_party/asr/${asr_ic_family}",
```

Comment:
if you already hardcode some things here, how about extracting the fixed "lwip/mbedtls-mbedtls-2.18/" from the list above as well?